### ISSUE TITLE
Anpassung Aufteilung der Wochentage

### ISSUE THREAD

**Issue by MMeinhardt1** [rg-engineering/ioBroker.heatingcontrol#155]
Hallo @rg-engineering,
ich persönlich finde die Aufteilung der Wochentage etwas unglücklich. Es ist vollkommen richtig das Sa und So die Tage des Wochenendes sind, allerdings gehen wir in der Regel Fr und Sa eher später ins Bett und So wieder etwas früher weils Montag wieder zur Arbeit geht.
Ich würde mir also eher eine Aufteilung So-Do + Fr-Sa wünschen anstatt Mo-Fr + Sa-So. 

Vielen Dank

**Comment by rg-engineering** [comment]
ja, auch diese Diskussion hatten wir schon hier und ich verstehe durchaus die Argumentation. Aber der nächste hat jeden Freitag frei und benötigt deshalb wieder andere Aufteilungen.
Ich hatte damals entschieden, solche "Spezialfälle" nicht zu integrieren, stattdessen dafür den Profiltype "jeden Tag einzeln" zu empfehlen...

**Comment by MMeinhardt1** [comment]
Ok, das kann ich ebenfalls nachvollziehen. Nur jeden Tag einzeln ist recht aufwändig. Gibt es denn ne programmiertechnische Möglichkeit es so zu gestalten das man sich die Wochentage die gleich sein sollen mit Checkboxen zusammenzufassen kann? Das dürfte dann ja allen individuellen Wünschen und Möglichkeiten entsprechen.

Danke und noch nen schönen Sonntag.

**Comment by rg-engineering** [comment]
Im Moment gibt es dieses Feature (noch) nicht... Die Anfrage gab es aber auch schon ein / zweimal...
Auf der anderen Seite, das Szenario, welches du oben beschreibst, benötigt einen Freitag (morgens wie Donnerstag, abends wie Samstag), einen Samstag und einen Sonntag (morgens wie Samstag und abends wie Montag)... Das sind schon mal drei verschiedene Tage...

**Comment by MMeinhardt1** [comment]
Hi, grundsätzlich hast Du recht. Bei mir wäre es allerdings so wie eingangs beschrieben, morgens alles gleich nur abends wäre der Unterschied. Also nur zwei verschiedene Tagesvarianten.
Dein beschriebenes Szenario wäre vom Aufwand aber immer noch geringer als das was jetzt möglich wäre, jeden Tag einzeln zu machen.
Aber Du bist derjenige der es machen müsste und wenn es gegen Deine Philosophie ist dann bleibt mir nur der eine Weg.

Dank Dir trotzdem.

**Comment by MMeinhardt1** [comment]
PS. Im ShutterControl Adapter hast Du es ja genauso angepasst wie ich es oben geschrieben habe, Freitag = Wochenende anstatt Sonntag 😉

**Comment by rg-engineering** [comment]
erwischt ;-)

**Comment by MMeinhardt1** [comment]
> erwischt ;-)
Das Internet weiß eben alles 😉
> 
> aber da betrifft es ja auch nur den Abend (=Herunterfahren)...
Das wäre für mich persönlich ausreichend 😃

**Comment by MMeinhardt1** [comment]
Hi @rg-engineering,
ich muss das Thema jetzt nochmal rausholen und nachfragen ob es in der V2 irgendwie möglich wäre das ganze anpassbar zu machen. In den Profilen unter Profiltyp sind zur Zeit 3 Profile auswählbar, wäre es denk- und machbar das es dort einen weiteren Profiltyp "Individuell" bei dem man dann die gewünschte Anzahl an Profilen eingibt und die Wochentage z.B. entsprechend wie im Bild gezeigt markieren kann?

Profil 1 ![hc1](https://user-images.githubusercontent.com/56205459/103307538-98e4dd80-4a10-11eb-9694-97ad26470f35.png)

Profil 2 ![hc](https://user-images.githubusercontent.com/56205459/103307518-8cf91b80-4a10-11eb-88e4-c7d98e9144e0.png)

Vielen Dank

**Comment by rg-engineering** [comment]
ich baue eine copy-Funktion ein, somit kann man Profile und Perioden kopieren und dann entsprechend einstellen...
Das oben gezeigte ist schön, aber derzeit vom Aufwand nicht machbar...

**Comment by Shine-** [comment]
Mir ist bewusst, dass dieses Thema bereits 4 Jahre alt ist und seitdem nicht weiter verfolgt wurde.

Dennoch, falls noch niemand darauf gekommen sein sollte, wie wäre es denn im Mo-Fr/Sa-Su Profilmodus beispielsweise mit Werten für:

"WeekendStartsEarlier" = x Stunden
"WeekendEndsEarlier" = y Stunden

Bei x = 10:00 h würde beispielsweise bereits Freitags ab 14:00 Uhr das Sa-Su Profil genutzt und bei y = 02:00 h würde beispielsweise ab Sonntag 22:00 Uhr wieder das Mo-Fr Profil genutzt.

Im Fall von "Feiertag wie Sonntag" könnte man den Datenpunkt ".morgen" im Feiertage-Adapter verwenden, um "x" Stunden vor einem Feiertag bereits das Sa-Su Profil gelten zu lassen und am Feiertag "y" Stunden vor Mitternacht wieder das auf den jeweiligen Feiertag folgende Profil, sei es nun Mo-Fr oder Sa-Su.

Würden die genannten Offsets, anstatt einer komplett neuen Wochentags-Logik, den Implementierungsaufwand soweit in Grenzen halten, dass es vertretbar ist?
